tests/main/snap-user-service-socket-activation/task.yaml

```diff
@@ -0,0 +1,47 @@
+summary: Check that sockets for user services are created correctly
+
+systems:
+    # The core snap currently doesn't set /etc/systemd/user as a
+    # writable path, so the test fails
+    - -ubuntu-core-*
+    # Ubuntu 14.04's systemd doesn't have user@.service
+    - -ubuntu-14.04-*
+    # Amazon Linux 2 gives error "Unit user@12345.service not loaded."
+    - -amazon-linux-2-*
+    # Centos 7 gives error "Unit user@12345.service not loaded."
+    - -centos-7-*
```

Comment:
What is the user experience for someone installing a snap with a user service on these distributions (or others user services aren't in use yet)?

Reply:
The problem is that the method of launching the user mode systemd has changed over the course of releases.

With current versions of systemd, the supported method is to ask the pid 1 systemd instance to start the templated `user@.service` service.  With older releases, my understanding is that `systemd --user` was invoked directly by logind.

As written, the test relies on the new method of launching the user instance, since trying directly execute it will fail with complaints about cgroups.  I haven't looked into what would be necessary to get the test fixture running on these old systemd's, and wasn't sure how important it was.

My primary motive for this feature is for use in conjunction with dbus service activation.  For that particular case, I plan to use the same method of supporting distros without a user instance of systemd as most services do currently: include both an `Exec` and `SystemdService` line in the dbus service activation file.

For other types of user services I don't have a good answer.  It is basically the same situation as running a snap that uses XDG Autostart on a desktop that doesn't support it.

Reply:
Thanks for the detailed response. Sounds good :)

Reply:
> Would using session-tool avoid the problem?

I suspect it wouldn't.  I booted up the Centos 7 live CD in a VM to see what its desktop session looked like, and there was no evidence of a user instance of systemd running.

Either it ships with a too-old systemd, or they intentionally disabled it so it wouldn't be part of their support burden.